### PR TITLE
MAILET-144 Fix SMIME decryption mailet

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

**Comment by aduprat** [comment]
test this please

File: mailet/crypto/src/main/java/org/apache/james/transport/mailets/SMIMEDecrypt.java [hunk 1]
```diff
@@ -100,7 +103,17 @@ public void init() throws MessagingException {
...
 
-        
+        certificateHolder = from(keyHolder.getCertificate());
+    }
+
+    private X509CertificateHolder from(X509Certificate certificate) throws MessagingException {
+        try {
+            return new X509CertificateHolder(certificate.getEncoded());
+        } catch (CertificateEncodingException e) {
+            throw new MessagingException("Error during the decryption of the certificate",e);
```
Comment (mbaechler) [inline mailet/crypto/src/main/java/org/apache/james/transport/mailets/SMIMEDecrypt.java]: is it decryption ?

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/TemporaryFilesystemModule.java [hunk 2]
```diff
@@ -54,7 +54,8 @@
             "recipientrewritetable.xml",
             "smtpserver.xml",
             "usersrepository.xml",
-            "smime_1.p12");
+            "smime_1.p12",
+            "smime.p12");
```
Comment (mbaechler) [inline server/mailet/integration-testing/src/test/java/org/apache/james/mailets/TemporaryFilesystemModule.java]: could we get some names to understand what is the role of each smime.p12 ?

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/crypto/SMIMEDecryptIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    private ConditionFactory calmlyAwait;
+
+    @Before
+    public void setup() throws Exception {
+        MailetContainer mailetContainer = MailetContainer.builder()
+            .postmaster("postmaster@" + DEFAULT_DOMAIN)
+            .threads(5)
+            .addProcessor(CommonProcessors.root())
+            .addProcessor(CommonProcessors.error())
+            .addProcessor(ProcessorConfiguration.builder()
```
Comment (mbaechler) [inline server/mailet/integration-testing/src/test/java/org/apache/james/mailets/crypto/SMIMEDecryptIntegrationTest.java]: why don't you use a simpler pipeline ? most of the mailers here are useless for the test we wrote

**Comment by aduprat** [comment]
test this please

File: server/mailet/integration-testing/src/test/java/org/apache/james/mailets/crypto/SMIMEDecryptIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    public void cryptedMessageShouldNotBeDecryptedWhenCertificateDoesntMatch() throws Exception {
+
+        try (SMTPMessageSender messageSender = SMTPMessageSender.authentication(LOCALHOST_IP, SMTP_SECURE_PORT, DEFAULT_DOMAIN, FROM, PASSWORD);
+             IMAPMessageReader imapMessageReader = new IMAPMessageReader(LOCALHOST_IP, IMAP_PORT)) {
+            messageSender.sendMessageWithHeaders(FROM, FROM, IOUtils.toString(ClassLoader.getSystemResourceAsStream("eml/bad_crypted.eml"))); 
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(messageSender::messageHasBeenSent);
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(() -> imapMessageReader.userReceivedMessage(FROM, PASSWORD));
+
+            assertThat(imapMessageReader.readFirstMessageInInbox(FROM, PASSWORD))
+                .containsSequence("MIAGCSqGSIb3DQEHA6CAMIACAQAxggKpMIICpQIBADCBjDCBhjELMAkGA1UE");
```
Comment (quynhn) [inline server/mailet/integration-testing/src/test/java/org/apache/james/mailets/crypto/SMIMEDecryptIntegrationTest.java]: What is the long string value?

Comment (aduprat) [inline server/mailet/integration-testing/src/test/java/org/apache/james/mailets/crypto/SMIMEDecryptIntegrationTest.java]: It's a part of the encrypted message content.
If the content is decrypted, the assertion will fail.

Comment (mbaechler) [inline server/mailet/integration-testing/src/test/java/org/apache/james/mailets/crypto/SMIMEDecryptIntegrationTest.java]: you can extract the magic value to a local variable

**Comment by guillaumebadin** [comment]
This mailet must be re-opened,
The smime decryption corrupt the attached files in the mails.

**Comment by aduprat** [comment]
#560

**Pull request closed without merging**